test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +103,51 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', { data: readable } as
+        Representation)));
+
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const drainData = new Promise((resolve): any => {
+      representation.data.on('data', (): any => readable);
+      representation.data.prependListener('end', (): any => {
+        order.push('end');
+        resolve();
+      });
+    });
+    await drainData;
+    expect(locker.acquire).toHaveBeenCalledTimes(1);
+    expect(locker.acquire).toHaveBeenLastCalledWith({ path: 'path' });
+    expect(source.getRepresentation).toHaveBeenCalledTimes(1);
+    expect(lock.release).toHaveBeenCalledTimes(1);
+    expect(order).toEqual([ 'acquire', 'getRepresentation', 'end', 'release' ]);
+  });
+
+  it('releases the lock on the resource when readable errors.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', { data: readable } as
+        Representation)));
+
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const drainDataAndHandleError = new Promise((resolve): any => {
+      representation.data.on('data', (): any => readable);
```

Comment:
Mentioned this in the previous thread, but in case that gets lost: this line should be removed. (Otherwise we don't even guarantee that the release has been called be there being an error).